code/modules/mob/language.dm

```diff
@@ -476,35 +476,55 @@
 		scramble_cache[input] = n
 		return n
 
-	var/input_size = length(input)
-	var/scrambled_text = ""
+	var/list/scrambled_text_pieces = list(input)
+	var/list/found_names = list()
+	for(var/mob/living/carbon/human/H in player_list)
+		var/list/nameparts = splittext(H.real_name," ")
+		if(nameparts.len > 3) // so the clown or borgs can't abuse this to reveal languages with common words
```

Comment:
I am glad you took this into consideration

More broadly though, I wonder if it would be better to just use our general name lists. Those cover general name usage and uncommon names are just that. By using the general human names list + some brands, for example, you would still know if someone is talking about John Nanotrasen even if no one is playing a John.

Reply:
realistically i would actually prefer if it was just names the character hearing it were deeply familiar with (particularly their own) but that seems a nightmare to code.

Reply:
I've seen a couple normal players with longer than 3 words names, those would be affected by the namelength check, like "Michael John Pemberton II" or similar. It's a perfectly valid name.

Reply:
so increase it to 4?

Reply:
Or 5. Maybe also have some specifics blacklisted, such as names that start with "clown that" or "mime that" so the "clown that kicks you and laughs" or "mime that hides from greyshirts" gimmicks don't see the effect.

Reply:
I still think the most logical way to approach this is to just do names from our standard name lists. Much less guesswork and much cleaner. Also incentivizes people to increase those lists with more reasonable human and alien names.

Reply:
That works for humans and certain other species, but most species generate their random names from lists of syllables, rather than lists of full names.
I don't really have any implementation suggestions because absolutely any possible implementation will have problems, by the very nature of language